lib/dep_graph.js

```diff
@@ -242,3 +242,26 @@ DepGraph.prototype = {
     }
   }
 };
+
+/**
+ * Cycle error, including the path of the cycle.
+ */
+var DepGraphCycleError = exports.DepGraphCycleError = function (cyclePath) {
+  var message = 'Dependency Cycle Found: ' + cyclePath.join(' -> ')
+  var instance = new Error(message);
+  instance.cyclePath = cyclePath;
+  Object.setPrototypeOf(instance, Object.getPrototypeOf(this));
+  if (Error.captureStackTrace) {
+    Error.captureStackTrace(instance, DepGraphCycleError);
+  }
+  return instance;
+}
+DepGraphCycleError.prototype = Object.create(Error.prototype, {
+  constructor: {
+    value: Error,
+    enumerable: false,
+    writable: true,
+    configurable: true
+  }
+});
+Object.setPrototypeOf(DepGraphCycleError, Error);
```

Comment:
[Custom errors in ES5] are more complicated than extending regular objects.

[1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error#Custom_Error_Types